### PR TITLE
Fix initialism errors (PR-3) 

### PR DESCRIPTION
**Pull request by haardikdharma10**
#### Summary

For #16623 partially (This is PR-3)

#### Ticket Link

#16623  

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/bot.go [hunk 1]
```diff
@@ -364,7 +364,7 @@ func (a *App) disableUserBots(userID string) *model.AppError {
 
 		// Get next set of bots if we got the max number of bots
 		if len(userBots) == perPage {
-			options.Page += 1
+			options.Page++
```
Comment (agnivade) [inline app/bot.go:367]: While this is good to do, but it does not give much gain in readability. Let's revert these changes.

File: plugin/client_rpc.go [hunk 2]
```diff
@@ -610,25 +610,25 @@ func (s *hooksRPCServer) MessageWillBeUpdated(args *Z_MessageWillBeUpdatedArgs,
 	return nil
 }
 
-type Z_LogDebugArgs struct {
+type ZLogDebugArgs struct {
```
Comment (agnivade) [inline plugin/client_rpc.go:613]: @crspeller - Will there be any issues with changing these exported structs? Are there consumers depending on these?

File: app/bot.go (hunk 1 above)
Comment (haardikdharma10) [reply app/bot.go:367]: Reverting it back to original will cause the lint error. Is it fine?

File: plugin/client_rpc.go (hunk 2 above)
Comment (crspeller) [reply plugin/client_rpc.go:613]: I think it should be fine. I think these are only exported for serialization (the reflect limitation). 
But I could be wrong. Someone should test to make sure plugins work on this branch. Break should be super obvious. 

Also a reminder that while the changes to this file will stick. The changes in  `plugin/client_rpc_generated.go` will not as that file is generated by `interface_generator`. That generator will have to be modified.

File: app/bot.go (hunk 1 above)
Comment (agnivade) [reply app/bot.go:367]: What lint error are you getting for this?

File: plugin/client_rpc.go (hunk 2 above)
Comment (agnivade) [reply plugin/client_rpc.go:613]: Oh wow, that's a great catch, @haardikdharma10 did you manually modify each and every line of `plugin/client_rpc_generated.go`? I totally missed the fact that `plugin/interface_generator/main.go` is not touched at all :facepalm: 

Let's change that and then generate the file.

Comment (haardikdharma10) [reply plugin/client_rpc.go:613]: Yes @agnivade! It was as simple as replacing all `Z_` with `Z`. However, I see that changing [this line](https://github.com/mattermost/mattermost-server/blob/master/plugin/interface_generator/main.go#L473) in `plugin/interface_generator/main.go` is all I have to do here. I have replaced it from `Z_ + name` to `Z + name`. How do i generate the `plugin/client_rpc_generated.go` file now?

Comment (crspeller) [reply plugin/client_rpc.go:613]: @haardikdharma10 `make pluginapi` should do it.

Comment (haardikdharma10) [reply plugin/client_rpc.go:613]: @crspeller can't figure out what's with the failing check? 🤔

**Comment by agnivade** [comment]
@jupenur - Is this a known failure https://app.circleci.com/pipelines/github/mattermost/mattermost-server/22418/workflows/fb29d59a-df05-49c9-8c86-75ae26f000cb/jobs/138831 ?

**Comment by jupenur** [comment]
@agnivade yeah, see https://community.mattermost.com/core/pl/14pjeddp4jgg8n63kpsbogcj8e

**Comment by agnivade** [comment]
Oh my bad, I thought that was a different one.

/update-branch

**Review by isacikgoz — APPROVED** [review]

**Comment by haardikdharma10** [comment]
Is this good enough to be merged? 

cc @agnivade, @isacikgoz

File: model/client4.go [hunk 3]
```diff
@@ -4148,7 +4148,7 @@ func (c *Client4) GetGroups(opts GroupSearchOpts) ([]*Group, *Response) {
 }
 
 // GetGroupsByUserId retrieves Mattermost Groups for a user
-func (c *Client4) GetGroupsByUserId(userId string) ([]*Group, *Response) {
+func (c *Client4) GetGroupsByUserID(userId string) ([]*Group, *Response) {
```
Comment (agnivade) [inline model/client4.go:4151]: Please revert this. Anything in the model package is off limits :P

**Review by agnivade — CHANGES_REQUESTED** [review]

File: model/client4.go (hunk 3 above)
Comment (agnivade) [reply model/client4.go]: You need to change the call sites too.

**Review by agnivade — APPROVED** [review]

**Comment by haardikdharma10** [comment]
All checks are passing 🎉 

@agnivade, @isacikgoz
